src/arch/xtensa/include/xtensa/xtruntime-frames.h

```diff
@@ -136,16 +136,30 @@ STRUCT_END(UserFrame)
  *
  * To access the core specific structure from ASM (after threadptr is set):
  * xtos_addr_percore a13, xtos_interrupt_table
- *
- * Access to the core specific structure from C is not supported!
  */
 STRUCT_BEGIN
 STRUCT_FIELD(void*,4,XTOS_PTR_TO_,xtos_enabled)
 STRUCT_FIELD(void*,4,XTOS_PTR_TO_,xtos_intstruct)
 STRUCT_FIELD(void*,4,XTOS_PTR_TO_,xtos_interrupt_table)
 STRUCT_FIELD(void*,4,XTOS_PTR_TO_,xtos_interrupt_mask_table)
+STRUCT_FIELD(void*,4,XTOS_PTR_TO_,xtos_stack_for_interrupt_1)
```

Comment:
The commit title could be more precise and use "dedicated stacks for interrupt handlers" instead of "new structs".

The commit body may mention this is required to implement the execution content switching. Which is useful for any pre-emptive scheduler and EDF is just one of potential options here.